### PR TITLE
Add curl check of the local instance

### PR DESCRIPTION
**Pull request by hnykda**

### CODE REVIEW

**Comment by hnykda** [comment]
@ThomDietrich WDYT so far? Something like this?

I was also thinking that a fancy way would be some kind of autodiscovery for beginners, e.g. using nmap, but that sounds too hard.

File: autossh/config.yaml [hunk 1]
```diff
@@ -21,15 +21,20 @@ options:
   hostname: ''
   ssh_port: 22
   username: autossh
-  remote_forwarding:
-    - '127.0.0.1:8123:172.30.32.1:8123'
+  forward_local_ip_address: "172.17.0.1"
```
Comment (ThomDietrich) [inline autossh/config.yaml:24]: Shouldn't that be this default?

```suggestion
  forward_local_ip_address: "172.30.232.1"
```

File: autossh/config.yaml [hunk 2]
```diff
@@ -21,15 +21,20 @@ options:
...
+  forward_local_port: 8123
+  forward_remote_ip_address: "127.0.0.1"
+  forward_remote_port: 8124
```
Comment (ThomDietrich) [inline autossh/config.yaml:27]: Let's shift remote and local in their order? I'm not sure whether `forward_` is needed?

File: autossh/config.yaml [hunk 3]
```diff
@@ -21,15 +21,20 @@ options:
...
+
   other_ssh_options: '-v -N'
   force_keygen: false
 schema:
   hostname: str
   ssh_port: int
   username: str
-  remote_forwarding:
-    - str
+  local_ip_address: str
```
Comment (ThomDietrich) [inline autossh/config.yaml:35]:
```suggestion
  local_ip_address: str?
```

File: autossh/config.yaml [hunk 4]
```diff
@@ -21,15 +21,20 @@ options:
...
+  local_port: int
```
Comment (ThomDietrich) [inline autossh/config.yaml:36]:
```suggestion
  local_port: int?
```


See https://developers.home-assistant.io/docs/add-ons/configuration/#options--schema

File: autossh/run.sh [hunk 5]
```diff
@@ -46,6 +51,15 @@ if [ -z "$HOSTNAME" ]; then
   exit 1
 fi
 
+local_forward_uri="$FORWARD_LOCAL_IP_ADDRESS:$FORWARD_LOCAL_PORT"
+bashio::log.info "Checking accessibility of $local_forward_uri"
+status_code=$(curl --write-out %{http_code} --silent --output /dev/null $local_forward_uri)
+if [[ "$status_code" -ne 200 ]] ; then
+  bashio::log.error "Check failed. $local_forward_uri resulted HTTP status_code=$status_code and not 200. Is the address correct?" && exit 1
+else
+  bashio::log.info "Check passed. $local_forward_uri returned HTTP status_code=200"
```
Comment (ThomDietrich) [inline autossh/run.sh:60]: I would suggest a less technical message like 
```suggestion
  bashio::log.info "Testing local port definition... Home Assistant web frontend can be reached"
```

**Comment by ThomDietrich** [comment]
Nice one! I've asked for a PR but didn't expect one within 24h :) Looks very good! I left a few comments.

File: autossh/config.yaml (hunk 3 above)
Comment (hnykda) [reply autossh/config.yaml]: Hm, what happens if this is not provided? 🤔 Just uses localhost?

File: autossh/run.sh [hunk 6]
```diff
@@ -8,7 +8,12 @@ HOSTNAME=$(jq --raw-output ".hostname" $CONFIG_PATH)
 SSH_PORT=$(jq --raw-output ".ssh_port" $CONFIG_PATH)
 USERNAME=$(jq --raw-output ".username" $CONFIG_PATH)
 
-REMOTE_FORWARDING=$(jq --raw-output ".remote_forwarding[]" $CONFIG_PATH)
+FORWARD_LOCAL_IP_ADDRESS=$(jq --raw-output ".forward_local_ip_address" $CONFIG_PATH)
+FORWARD_LOCAL_PORT=$(jq --raw-output ".forward_local_port" $CONFIG_PATH)
+FORWARD_REMOTE_IP_ADDRESS=$(jq --raw-output ".forward_remote_ip_address" $CONFIG_PATH)
+FORWARD_REMOTE_PORT=$(jq --raw-output ".forward_remote_port" $CONFIG_PATH)
+REMOTE_FORWARDING="$FORWARD_REMOTE_IP_ADDRESS:$FORWARD_REMOTE_PORT:$FORWARD_LOCAL_IP_ADDRESS:$FORWARD_LOCAL_PORT"
```
[the remark this answers is not in the document]
Comment (hnykda) [reply autossh/run.sh:15]: Done. I have kept FORWARD_ in the name to make it clear this is not hostname.

File: autossh/run.sh (hunk 5 above)
Comment (hnykda) [reply autossh/run.sh]: Done, good thinking. I kept the error one a bit more technical to provide some additional debug info e.g. in the cases people of some forwarding turned on (301 will now fail).

**Comment by hnykda** [comment]
Can you resolve pending comments if they are OK please?

**Comment by hnykda** [comment]
FYI I haven't tested it yet, so keeping it in draft. Will try to test it when I am at home and can be sure I will not lock myself out :-D ...

Is changing https://github.com/hnykda/home-assistant-addons/blob/patch-1/repository.json#L2-L4 to something else, then adding it in https://<my-ha>/hassio/store as a new repo and then testing it enough?

**Comment by ThomDietrich** [comment]
Hey! Did you want to continue with this? I am looking forward to this improvement

**Comment by hnykda** [comment]
Hey! I am sorry, I haven't managed to get back to it, unfortunately 😞 . Hm, can you take it from me or is it unlikely?

**Review by ThomDietrich — APPROVED** [review]

**Review by ThomDietrich — APPROVED** [review]

**Pull request merged**